server/services/store/sqlstore/user.go

```diff
@@ -27,26 +29,45 @@ func (s *SQLStore) GetRegisteredUserCount() (int, error) {
 }
 
 func (s *SQLStore) getUserByCondition(condition sq.Eq) (*model.User, error) {
-	query := s.getQueryBuilder().
-		Select("id", "username", "email", "password", "mfa_secret", "auth_service", "auth_data", "props", "create_at", "update_at", "delete_at").
-		From(s.tablePrefix + "users").
-		Where(sq.Eq{"delete_at": 0}).
-		Where(condition)
-	row := query.QueryRow()
-	user := model.User{}
-
-	var propsBytes []byte
-	err := row.Scan(&user.ID, &user.Username, &user.Email, &user.Password, &user.MfaSecret, &user.AuthService, &user.AuthData, &propsBytes, &user.CreateAt, &user.UpdateAt, &user.DeleteAt)
+	users, err := s.getUsersByCondition(condition)
 	if err != nil {
 		return nil, err
 	}
 
-	err = json.Unmarshal(propsBytes, &user.Props)
+	var user *model.User
+
+	if len(users) > 0 {
+		user = users[0]
+	}
```

Comment:
Previously `.QueryRow` was used which accord to documentation, throws an error if no records are returned. 
https://golang.org/pkg/database/sql/#DB.QueryRow
`.Query` does not throw an error when no rows are returned. This may be find depending on how calling function are working, but thought I'd mention it.

Reply:
Great catch! The function's callers are actually assuming a non-nil return value when there is no error. I've fixed this now by manually returning`sql.ErrNoRows` when no data is found.